### PR TITLE
Document PERLIO_FUNCS_(DECL|CAST)

### PR DESCRIPTION
**Pull request by khwilliamson**

### CODE REVIEW

**Review by leonerd — APPROVED** [review]
LGTM

**Comment by Leont** [comment]
It's actually wrong. It doesn't declare a PerlIO function, but a PerlIO virtual function table

File: perlio.h [hunk 1]
```diff
@@ -63,6 +63,16 @@ typedef PerlIOl *PerlIO;
 #define PerlIO PerlIO
 #define PERLIO_LAYERS 1
 
+/*
+=for apidoc_section $io
+=for apidoc Amu||PERLIO_FUNCS_DECL|PerlIO * func
+Declare C<func> to be a PerlIO function, that is, of type C<PerlIO_funcs>.
```
Comment (Leont) [inline perlio.h]: Declare C&lt;func> to be a PerlIO function table, that is, of type C<PerlIO_funcs>.

File: perlio.h [hunk 2]
```diff
@@ -63,6 +63,16 @@ typedef PerlIOl *PerlIO;
...
+
+=for apidoc Ay|PerlIO_funcs *|PERLIO_FUNCS_CAST|PerlIO * func
+Cast the function C<func> to be of type S<C<PerlIO_funcs *>>.
```
Comment (Leont) [inline perlio.h]: Cast the pointer C&lt;func> to be of type S<C<PerlIO_funcs *>>.

**Pull request merged**